code/game/objects/items/weapons/implant/implants/carrion/holo_spider.dm

```diff
@@ -0,0 +1,237 @@
+		desc = initial(desc)
+		icon = initial(icon)
+		icon_state = initial(icon_state)
+		if(is_equipped(src))
+			layer = ABOVE_HUD_LAYER
+			plane = ABOVE_HUD_PLANE
+		else
+			layer = initial(layer)
+			plane = calculate_plane(z, original_plane)
+		item_state = initial(item_state)
+		set_dir(initial(dir))
+		update_icon()
+		to_chat(owner_mob, SPAN_NOTICE("You deactivate the [src]."))
+	else
+		if(!saved_item)
+			to_chat(owner_mob, SPAN_NOTICE("The [src] does not have anything scanned."))
+			return
+		else
+			activate_holo(saved_name, saved_icon, saved_icon_state, saved_description, saved_dir, saved_appearance, saved_item_state)		
+			to_chat(owner_mob, SPAN_NOTICE("You activate the [src]."))
+
+/obj/item/implant/carrion_spider/holographic/proc/activate_holo(new_name, new_icon, new_iconstate, new_description, new_dir, new_appearance, new_item_state)
+	name = new_name
+	desc = new_description
+	icon = new_icon
+	icon_state = new_iconstate
+	item_state = new_item_state
+	appearance = new_appearance
+	set_dir(new_dir) 
+	if(is_equipped(src))
+		plane = ABOVE_HUD_PLANE
+		layer = ABOVE_HUD_LAYER
+	else
+		plane = calculate_plane(z, saved_original_plane)
+		layer = saved_layer
+	dummy_active = TRUE
+
+/obj/item/implant/carrion_spider/holographic/proc/reset_data()
+	saved_name = initial(saved_name)
+	saved_item = initial(saved_item)
+	saved_type = initial(saved_type)
+	saved_icon = initial(saved_icon)
+	saved_icon_state = initial(saved_icon_state)
+	saved_description = initial(saved_description)
+	saved_dir = initial(saved_dir)
+	saved_message = initial(saved_message)
+	saved_appearance = initial(appearance)
+	saved_item_state = initial(item_state)
+	saved_w_class = initial(saved_w_class)
+	saved_gender = initial(saved_gender)
+	saved_layer = initial(saved_layer)
+	saved_original_plane = initial(saved_original_plane)
+
+/obj/item/implant/carrion_spider/holographic/dropped(mob/living/W)
+	if(dummy_active)
+		layer = saved_layer
+		plane = calculate_plane(z, saved_original_plane)
+	else
+		layer = initial(layer)
+		plane = calculate_plane(z, original_plane)
+	..()
+
+
+/obj/item/implant/carrion_spider/holographic/examine(mob/user, var/distance = -1)
+	if(dummy_active && saved_item && saved_message)
+		to_chat(user, saved_message)
+	else if(dummy_active && saved_item)
+		var/message
+		var/size
+		if(istype(saved_item, /obj/item))
+			switch(saved_w_class)
+				if(ITEM_SIZE_TINY)
+					size = "tiny"
+				if(ITEM_SIZE_SMALL)
+					size = "small"
+				if(ITEM_SIZE_NORMAL)
+					size = "normal-sized"
+				if(ITEM_SIZE_BULKY)
+					size = "bulky"
+				if(ITEM_SIZE_HUGE)
+					size = "huge"
+				if(ITEM_SIZE_GARGANTUAN)
+					size = "gargantuan"
+				if(ITEM_SIZE_COLOSSAL)
+					size = "colossal"
+				if(ITEM_SIZE_TITANIC)
+					size = "titanic"
+			message += "\nIt is a [size] item."
+			var/obj/item/N = saved_item
+			for(var/Q in N.tool_qualities)
+				message += "\n<blue>It possesses [tool_qualities[Q]] tier of [Q] quality.<blue>"
+			qdel(N)
```

Comment:
Don't think you need qdel here, as N is only a reference to an object, not a new instanced object.